### DISCUSSION TITLE
Dependencia Circular Entity Framework

### DISCUSSION THREAD

**Discussion by IgorCunha1** [balta-io/forum#312] [Tira Dúvidas] [answered]
Estou fazendo uma consulta com o EF e está retornando Circular Dependency pois é um relacionamento muitos p/ muitos, não criei o Identity que une os relacionamentos entre eles.
` public class Servico
    {

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Descricao { get; set; }

               
        public List<Empresa> Empresa { get; set; }
        public List<Agendamento> Agendamento { get; set; }

    }`
    
    `[Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Nome { get; set;}
        public string RazaoSocial { get; set;}
        public string Cnpj { get; set;}
        public string Email { get; set;}
        public string Telefone { get; set;}

        [ForeignKey("EnderecoId")]
        public int EnderecoId { get; set; }
                        
        public Endereco Endereco { get; set; }
        public List<Servico> Servico { get; set; }`

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @IgorCunha1, como vai?

Qual mensagem de erro você recebe e onde ela ocorre?

**Comment by IgorCunha1** [comment]
Boa tarde Baltieri.
Desde já agradeço a ajuda, estou gostando muito de seus cursos.
Sei que está acontecendo o erro porque no model Servico eu carrego uma Lista de Empresas que também carrega uma lista de Servico.
A dúvida é, como resolver isso? 
no video vc mostrou como fazer isso mapeando todos os models pelo EF, 
mas também falou que tem como resolver isso de outra maneira, é dessa segunda forma que estou procurando.

Print Erro:
[image](https://user-images.githubusercontent.com/24699755/232819986-c9c89bb9-884e-47bd-8fec-f2ead8e593a0.png)
Codigo:
`
await _context.Servico
                                    .AsNoTracking()
                                    .Include(s => s.Empresa)
                                    .FirstOrDefaultAsync(s => s.Id == ServicoId);
`

**Reply by andrebaltieri** [reply]
Bom dia, @IgorCunha1 como vai?
Mesmo habilitando as opções de referência circular no Program.cs como mostro no curso de ASP.NET este erro ainda continua?

**Reply by IgorCunha1** [reply]
vou dar uma olhada aqui

**Reply by IgorCunha1** [reply]
Oi Baltieri muito obg pela dica.
Não sei porque mas não lembrei desse grande detalhe.

**Reply by andrebaltieri** [reply]
Tranquilo, acontece!!!
